lib/node_modules/@stdlib/_tools/remark/plugins/remark-lint-html-section-structure/lib/main.js

```diff
@@ -0,0 +1,366 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var logger = require( 'debug' );
+var rule = require( 'unified-lint-rule' );
+var visit = require( 'unist-util-visit' );
+var replace = require( '@stdlib/string/replace' );
+var trim = require( '@stdlib/string/trim' );
+
+
+// VARIABLES //
+
+var debug = logger( 'remark-lint-html-section-structure' );
+var SECTION_START = /<section(?:\s+class="([^"]*)")?>/;
+var SECTION_END = /<\/section>$/;
+var SECTION_END_WITH_COMMENT = /<\/section>\s*<!--\s*\/\.([^*]*)\s*-->/;
+var END_COMMENT_SIMPLE = /<!--\s*\/\.([^>]+)-->/;
+var END_COMMENT = /^\s*<!--\s*\/\.([^\s]*)\s*-->/;
+var htmlSectionStructureRule;
+
+
+// FUNCTIONS //
+
+/**
+* Creates a position string from a node's position.
+*
+* @private
+* @param {Object} node - AST node
+* @returns {string} position string in format "line:column"
+*/
+function formatPosition( node ) {
+	return node.position.start.line + ':' + node.position.start.column;
+}
+
+/**
+* Reports an error message.
+*
+* @private
+* @param {Object} file - virtual file
+* @param {Object} node - AST node to attach message to
+* @param {string} message - error message
+* @param {string} errorCode - error code identifier
+*/
+function reportErr( file, node, message, errorCode ) {
+	var fullMsg = formatPosition( node ) + '   error   ' + message + '   ' + errorCode;
+	debug( fullMsg );
+	file.message( fullMsg, node );
+}
+
+/**
+* Extracts comment class name from a node.
+*
+* @private
+* @param {Object} node - AST node containing comment
+* @param {RegExp} matchRegex - regular expression for matching comment
+* @param {RegExp} simpleMatchRegex - alternative regular expression for matching comment
+* @returns {string|null} extracted class name or null if no match
+*/
+function extractCommentClass( node, matchRegex, simpleMatchRegex ) {
+	var simpleMatch;
+	var match;
+
+	match = matchRegex.exec( node.value );
```

Comment:
Do we need to worry about resetting the match index since we are using `exec` here and below?

Reply:
No, we don't have global or sticky flag on the regular expressions that are executed here.